lib/houston/client.rb

```diff
@@ -80,5 +80,16 @@ def unregistered_devices
     def devices
       unregistered_devices.collect{|device| device[:token]}
     end
+
+    def certificate_data
+      return unless ENV['APN_CERTIFICATE']
+      if File.exist?(ENV['APN_CERTIFICATE'])
+        File.read(ENV['APN_CERTIFICATE'])
+      else
+        # Fallback and return the certificate itself; if it isn't readable, the
+        #   env var is likely the whole string version of the cert's `.pem`.
+        ENV['APN_CERTIFICATE']
```

Comment:
Right now this just assumes that if it's not a valid file path that the person intended it as a streaming/IO object. If you're going the route of taking it as is I'd put some other kind of validation around it.